src/test/java/uk/gov/pay/adminusers/service/NotificationServiceTest.java

```diff
@@ -0,0 +1,131 @@
+    
+    private static final String INVITE_USER_EMAIL_TEMPLATE_ID = "invite-user-email-template-id";
+    private static final String INVITE_USER_EXISTING_EMAIL_TEMPLATE_ID = "invite-user-existing-email-template-id";
+    private static final String FORGOTTEN_PASSWORD_EMAIL_TEMPLATE_ID = "forgotten-password-email-template-id";
+    
+    @Mock private NotifyClientProvider mockNotifyClientProvider;
+    @Mock private NotifyConfiguration mockNotifyConfiguration;
+    @Mock private NotifyDirectDebitConfiguration mockNotifyDirectDebitConfiguration;
+    @Mock private MetricRegistry mockMetricRegistry;
+
+    @Mock private NotificationClient mockNotificationClient;
+    @Mock private SendSmsResponse mockSendSmsResponse;
+
+    private NotificationService notificationService;
+
+    @Before
+    public void setUp() {
+        given(mockNotifyConfiguration.getSecondFactorSmsTemplateId()).willReturn(SECOND_FACTOR_SMS_TEMPLATE_ID);
+        given(mockNotifyConfiguration.getSignInOtpSmsTemplateId()).willReturn(SIGN_IN_OTP_SMS_TEMPLATE_ID);
+        given(mockNotifyConfiguration.getChangeSignIn2faToSmsOtpSmsTemplateId()).willReturn(CHANGE_SIGN_IN_2FA_TO_SMS_OTP_SMS_TEMPLATE_ID);
+        given(mockNotifyConfiguration.getSelfInitiatedCreateUserAndServiceOtpSmsTemplateId())
+                .willReturn(SELF_INITIATED_CREATE_USER_AND_SERVICE_OTP_SMS_TEMPLATE_ID);
+        given(mockNotifyConfiguration.getCreateUserInResponseToInvitationToServiceOtpSmsTemplateId())
+                .willReturn(CREATE_USER_IN_RESPONSE_TO_INVITATION_TO_SERVICE_OTP_SMS_TEMPLATE_ID);
+
+        given(mockNotifyConfiguration.getInviteUserEmailTemplateId()).willReturn(INVITE_USER_EMAIL_TEMPLATE_ID);
+        given(mockNotifyConfiguration.getInviteUserExistingEmailTemplateId()).willReturn(INVITE_USER_EXISTING_EMAIL_TEMPLATE_ID);
+        given(mockNotifyConfiguration.getForgottenPasswordEmailTemplateId()).willReturn(FORGOTTEN_PASSWORD_EMAIL_TEMPLATE_ID);
+        
+        given(mockNotifyClientProvider.get(CARD)).willReturn(mockNotificationClient);
+
+        given(mockMetricRegistry.histogram("notify-operations.sms.response_time")).willReturn(mock(Histogram.class));
+        
+        given(mockSendSmsResponse.getNotificationId()).willReturn(NOTIFICATION_ID);
+
+        notificationService = new NotificationService(mockNotifyClientProvider, mockNotifyConfiguration, mockNotifyDirectDebitConfiguration,
+                mockMetricRegistry);
+    }
+
+    @Test
+    public void sendSecondFactorPasscodeSmsWithSignInTemplate() throws NotificationClientException {
+        given(mockNotificationClient.sendSms(SIGN_IN_OTP_SMS_TEMPLATE_ID, PHONE_NUMBER_E164, Map.of("code", OTP), null))
+                .willReturn(mockSendSmsResponse);
+
+        String notificationId = notificationService.sendSecondFactorPasscodeSms(PHONE_NUMBER, OTP, OtpNotifySmsTemplateId.SIGN_IN);
+        
+        assertThat(notificationId, is(NOTIFICATION_ID.toString()));
+    }
```

Comment:
I think the real test should be verifying that `mockNotificationClient` is called with the correct template id, phone number and code. I can see this is effectively tested here but in a more long winded way.
It doesn't make sense to assert the `notificationId` (line 86).

Reply:
Yeah, I wasn’t too happy with this too. The fact that the the `NotificationClient` returns something pushed me towards structuring the test in a way that hides what’s really important. I’ll change it to verify the call to the `NotificationClient` and not worry about the return value.